### PR TITLE
systemctl: fix bus transport and runtime scope handling

### LINKED ISSUE #33661: OSS-Fuzz issue 70153
OSS-Fuzz has found a bug in this project. Please see https://oss-fuzz.com/testcase?key=5729581726236672 for details and reproducers.

This issue is mirrored from https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=70153 and will auto-close if the status changes there.

If you have trouble accessing this report, please file an issue at https://github.com/google/oss-fuzz/issues/new.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #33661.

### CODE REVIEW

**Review by YHNdnzj — APPROVED** [review]

File: src/systemctl/systemctl-logind.c [hunk 1]
```diff
@@ -432,6 +432,12 @@ int help_boot_loader_entry(void) {
         sd_bus *bus;
         int r;
 
+        /* This is called without checking runtime scope and bus transport like we do in parse_argv().
```
Comment (bluca) [inline src/systemctl/systemctl-logind.c:435]: shouldn't the combination be rejected instead?

Comment (yuwata) [reply src/systemctl/systemctl-logind.c:435]: This is just for help command, and provide some suggestions. I think it is not necessary to handle transport or scope options strictly.

Comment (bluca) [reply src/systemctl/systemctl-logind.c:435]: yeah but it gives the impression that it's a valid combination, which usually doesn't happen?

Comment (yuwata) [reply src/systemctl/systemctl-logind.c:435]: The `--boot-loader-entry=` positional argument sets `arg_boot_loader_entry`, and it is only used by `halt`, `poweroff`, and `reboot` commands. They are only supported by system scope.

So, even if a user get list of entries by `systemctl --user --boot-loader-entry=help`, we will refuse the operation by `systemctl --user --boot-loader-entry=foo reboot` with sufficient error message. Note the refusal is _not_ caused by `--boot-loader-entry=foo`, but by calling `reboot` with `--user`. And, obtained boot entries themselves obtained with adjusted scope and bus transport _are_ correct.

Comment (yuwata) [reply src/systemctl/systemctl-logind.c:435]: Upgrading the green label again. But feel free to downgrade it if you still do not like the current approach.

**Comment by bluca** [comment]
```
Assertion '!host' failed at src/shared/bus-util.c:434, function bus_connect_transport(). Aborting.
==320== ERROR: libFuzzer: deadly signal
    #0 0x55e9b5adc329 in __sanitizer_print_stack_trace /src/llvm-project/compiler-rt/lib/msan/msan.cpp:733:3
    #1 0x55e9b5c01928 in fuzzer::PrintStackTrace() /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerUtil.cpp:210:5
    #2 0x55e9b5be4653 in fuzzer::Fuzzer::CrashCallback() /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:231:3
    #3 0x55e9b5b31fe7 in SignalAction(int, void*, void*) /src/llvm-project/compiler-rt/lib/msan/msan_interceptors.cpp:1126:3
    #4 0x7f19075b641f  (/lib/x86_64-linux-gnu/libpthread.so.0+0x1441f) (BuildId: 9a65bb469e45a1c6fbcffae5b82a2fd7a69eb479)
    #5 0x7f190650200a in raise (/lib/x86_64-linux-gnu/libc.so.6+0x4300a) (BuildId: 0702430aef5fa3dda43986563e9ffcc47efbd75e)
    #6 0x7f19064e1858 in abort (/lib/x86_64-linux-gnu/libc.so.6+0x22858) (BuildId: 0702430aef5fa3dda43986563e9ffcc47efbd75e)
    #7 0x7f1906e524e4 in log_assert_failed /work/build/../../src/systemd/src/basic/log.c:996:9
    #8 0x7f1906e5974e in log_assert_failed_return /work/build/../../src/systemd/src/basic/log.c:1015:17
    #9 0x7f1906b313a0 in bus_connect_transport /work/build/../../src/systemd/src/shared/bus-util.c
    #10 0x55e9b5bb2e0e in acquire_bus /work/build/../../src/systemd/src/systemctl/systemctl-util.c:55:29
    #11 0x55e9b5b799c8 in help_boot_loader_entry /work/build/../../src/systemd/src/systemctl/systemctl-logind.c:441:13
    #12 0x55e9b5bc49e0 in systemctl_parse_argv /work/build/../../src/systemd/src/systemctl/systemctl.c:885:37
    #13 0x55e9b5b41dba in LLVMFuzzerTestOneInput /work/build/../../src/systemd/src/systemctl/fuzz-systemctl-parse-argv.c:58:13
    #14 0x55e9b5be5b70 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:614:13
    #15 0x55e9b5be5395 in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned long, bool, fuzzer::InputInfo*, bool, bool*) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:516:7
    #16 0x55e9b5be7322 in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::__Fuzzer::vector<fuzzer::SizedFile, std::__Fuzzer::allocator<fuzzer::SizedFile>>&) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:829:7
    #17 0x55e9b5be7657 in fuzzer::Fuzzer::Loop(std::__Fuzzer::vector<fuzzer::SizedFile, std::__Fuzzer::allocator<fuzzer::SizedFile>>&) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:867:3
    #18 0x55e9b5bd5c66 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:914:6
    #19 0x55e9b5c02192 in main /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerMain.cpp:20:10
    #20 0x7f19064e3082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082) (BuildId: 0702430aef5fa3dda43986563e9ffcc47efbd75e)
    #21 0x55e9b5ab199d in _start (build-out/fuzz-systemctl-parse-argv+0x6f99d)
```

**Pull request merged**